packages/flutter/lib/src/widgets/image_filter.dart

```diff
@@ -23,20 +28,34 @@ class ImageFiltered extends SingleChildRenderObjectWidget {
   /// The [imageFilter] must not be null.
   const ImageFiltered({
     Key? key,
-    required this.imageFilter,
+    this.imageFilter,
+    this.imageFilterCallback,
     Widget? child,
-  }) : assert(imageFilter != null),
+  }) : assert(imageFilter != null || imageFilterCallback != null,
+              'One of imageFilter or imageFilterCallback should be non-null'),
+       assert(imageFilter == null || imageFilterCallback == null,
+              'Only one of imageFilter or imageFilterCallback should be non-null'),
```

Comment:
fwiw, you can do this as a single assert by doing `(imageFilter == null) != (imageFilterCallback == null)`.
(feel free to ignore if you prefer your asserts, they're perfectly fine too)

Reply:
I've used that kind of condition before, but I thought it would be better to call out the conditions individually for debugging...?

(Granted, I'm probably more concerned that my "similar but different" assertion messages might be confusing than I am about the condition that triggers them.)

I'm not strongly attached to either, but I feel like the `((==) != (==))` pattern puts efficiency over being plain to subsequent maintainers and this is code that is only used for debug.

Reply:
Fixed in the latest commit. I used the more compact `(==) != (==)` assertion in the RenderObject but kept the separate asserts in the Widget so that they would generate more specific human-readable feedback.